editor/src/components/navigator/navigator-item/component-picker.tsx

```diff
@@ -293,62 +442,45 @@ const ComponentPickerComponentSection = React.memo(
         }}
         onScroll={onScroll}
       >
-        {components.map((component) => {
-          const selectedStyle =
-            component.value.key === currentlySelectedKey
-              ? {
-                  background: '#007aff',
-                  color: 'white',
-                }
-              : {}
-
-          return (
-            <FlexRow
-              css={{}}
-              key={component.value.key}
-              style={{
-                marginLeft: 8,
-                marginRight: 8,
-                borderRadius: 4,
-                // indentation!
-                paddingLeft: 8,
-                pointerEvents: isScrolling ? 'none' : 'auto',
-                color: '#EEE',
-                ...selectedStyle,
-              }}
-              onClick={onItemClick(component.value)}
-              onMouseOver={onItemHover(component.value)}
-              data-key={component.value.key}
-            >
-              <FlexRow css={{ gap: 10, height: 28, alignItems: 'center' }}>
-                <Icn
-                  {...iconPropsForIcon(component.value.icon ?? 'component')}
-                  width={12}
-                  height={12}
-                />
-                <label>{component.label}</label>
+        {components.reduce((acc, category) => {
+          const categoryMarkdown = category.items.map((component, index) => {
```

Comment:
Looks like a `flatMap` would simplify it a bit here

Reply:
Never used it before! Thanks for the suggestion.